Lib9c/BlockPolicy.cs

```diff
@@ -162,6 +162,16 @@ private InvalidBlockException ValidateMinerAuthority(Block<NCAction> block)
                     miner
                 );
             }
+            
+            // To prevent selfish mining, we define a consensus that blocks with no transactions are do not accepted. 
+            if (block.Transactions.Count <= 0)
+            {
+                return new InvalidMinerException(
+                    $"The block #{block.Index} {block.Hash}'s miner {miner} should be proven by " +
+                    "including a least one of transaction.",
```

Comment:
As it is intended to validate blocks, not miners…

```suggestion
                    $"The block #{block.Index} {block.Hash} (mined by {miner}) should " +
                    "include a least one of transaction.",
```

Reply:
Could you also change it to "at least one transaction"? :pray:

Reply:
https://github.com/planetarium/lib9c/pull/462/commits/3c20bc68d25860ab74eba48e20fb1983d77c29b2 I fixed it